sqllogictest-bin/src/lib.rs

```diff
@@ -526,30 +526,67 @@ async fn update_test_file<T: std::io::Write, D: AsyncDB>(
         let filename = filename.as_ref();
         let outfilename = filename.file_name().unwrap().to_str().unwrap().to_owned() + ".temp";
         let outfilename = filename.parent().unwrap().join(&outfilename);
-        let outfile = File::create(&outfilename)?;
+        // create a temp file in read-write mode
+        let outfile = OpenOptions::new()
+            .write(true)
+            .create(true)
+            .truncate(true)
+            .read(true)
+            .open(&outfilename)?;
         Ok((outfilename, outfile))
     }
 
+    fn override_with_outfile(
+        filename: &String,
+        outfilename: &PathBuf,
+        outfile: &mut File,
+    ) -> std::io::Result<()> {
+        // check whether outfile ends with multiple newlines, which happens if
+        // - the last record is statement/query
```

Comment:
cc @alamb This is an egde case we need to consider after adding `\n` to `Display` of statement and query. 😇

We need to consider it if we want file to end with newline. i.e., like
```
for record in records {
    println!("{record}");
}
```

In the roundtrip parsing test, it uses `join`